src/main/java/com/acrolinx/proxy/AcrolinxProxyHttpServlet.java

```diff
@@ -103,8 +122,8 @@ private static void transferTo(InputStream inputStream, OutputStream outputStrea
   private String genericToken;
   private String username;
 
-  public AcrolinxProxyHttpServlet() {
-    closeableHttpClient = createHttpClient();
+  public AcrolinxProxyHttpServlet(HttpServletTimeoutsConfig httpServletTimeoutsConfig) {
```

Comment:
While this HttpServletTimeoutsConfig is a good solution in most "normal" cases, I am not sure that this is the correct solution in case of an HttpServlet implementation like this class here. The class hierarchy is:

AcrolinxProxyHttpServlet > HttpServlet > GenericServlet > Servlet

Servlet is an interface and defines a getServletConfig() method which returns a javax.servlet.ServletConfig instance.

Should we not use this standard configuration mechanism instead?

Reply:
That is indeed a problem. Maybe we could work-around it by not setting the two timeouts as default request config when we construct the CloseableHttpClient, but instead set them on the HttpRequestBase instance each time inside the proxyRequest method?

Reply:
@Thilo-Acrolinx Now the tests are failing with SocketException, fixing that now!

Reply:
@Thilo-Acrolinx 
This SocketException behavior (both with enabled and disabled retries) is very undeterministic. Right now, the tests are passing and the connection has not been reset. If you try to re-run the tests, you will most probably receive a SocketException (with retries enabled) even though the port number is fixed (5500). I am not sure what this could be related to. :(

Reply:
Locally, the tests always pass.

Reply:
The problem is that SocketException is not the exception that we are expecting in that case. In the CI, that exception is not always thrown and that is where my confusion stems from.

Reply:
The reason why I changed the port to 5500 is because I thought that the random port could be the reason why the tests would pass fail sometimes, but that didn't prove to be the reason.